dataland-backend/src/test/kotlin/org/dataland/datalandbackend/controller/MetaDataControllerTest.kt

```diff
@@ -0,0 +1,90 @@
+package org.dataland.datalandbackend.controller
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.http.MediaType
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers.content
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
+
+@SpringBootTest
+@AutoConfigureMockMvc
+internal class MetaDataControllerTest(
+    @Autowired var mockMvc: MockMvc,
+    @Autowired var objectMapper: ObjectMapper
+) {
+
+    val testCompanyName = "Imaginary-Company_I"
+
+    @Test
+    fun `list of meta info about data for specific company can be retrieved`() {
+        CompanyUploader().uploadCompany(mockMvc, objectMapper, testCompanyName)
+
+        mockMvc.perform(
+            get("/metadata?companyId=1")
+                .accept(MediaType.APPLICATION_JSON)
+                .contentType(MediaType.APPLICATION_JSON)
+        )
+            .andExpectAll(
+                status().isOk,
+                content().contentType(MediaType.APPLICATION_JSON),
+                content().string("[]")
+            )
+    }
+
+/* The following tests require, that data is posted. To post data, a running instance of edc-dummyserver is needed.
+Until now, we haven't mocked the edc-dummyserver, and therefore the following unit tests cannot run.
+They stay commented out, until a decision is made.
```

Comment:
haben wir das nicht inzwischen?

Und: es fehlt noch ein Test für den EuTaxonomyDataController

Reply:
Wir haben das inzwischen, ja. Hamid und ich haben uns aber entschieden, hier trotzdem keine Tests zu implementieren. Gründe wie auch in dem anderen Kommentar:

1.) Die Sachen werden in den e2e-Tests sowieso getestet und sind somit gecovert.

2.) Vorher waren Unit-Tests, die Datenspeicherung erfordern, einfach umzusetzen, weil die Daten in den InMemoryDataStore geschrieben wurden. Jetzt werden Daten aber auf den edc-Dummyserver geschrieben.
D.h. beim Speichern findet eine Kommunikation zwischen Backend und einem Server statt.
Das über Mocks abzubilden/zu simulieren, ist bestimmt irgendwie möglich (auch wenn wir zwei nicht wissen wie).
Aber: Warum sich die Mühe machen, wenn es keinen Mehrwert bringt (siehe Punkt 1)?

Reply:
ok